examples/05_vol_geom_wasm_threejs/index.html

```diff
@@ -0,0 +1,330 @@
+		}
+
+		console.log("init done");
+		init_done = true;
+
+		return true;
+	}
+
+	function render() {
+		renderer.render(scene, camera);
+	}
+
+	Module['onRuntimeInitialized'] = function() {
+		console.log('WASM for VOL_GEOM started...');
+
+		init();
+		if (!init) { return false; }
+
+		document.getElementById("load_vols_files_button").hidden = false;
+		document.getElementById("load_vols_files_button").addEventListener("click", load_clicked, false);
+
+		console.log("render starting...");
+		renderer.setAnimationLoop(render);
+	}
+
+	// Copy vol_geom frame `frame_idx` into vologram webgl mesh.
+	// Returns true on success, and false on error loading frame.
+	// This function assumes that any preceding, required, keyframe has been resolved and loaded already.
+	function mesh_from_frame(frame_idx) {
+		if (vologram.last_frame_loaded == frame_idx) { return; } // Safety catch to avoid reloading the same frame twice.
+
+		// Ask the vol_geom WASM to read the frame data from the vologram file into `_frame_data`.
+		var ret = Module.ccall('read_frame', 'boolean', ['number'], [frame_idx]);
+		if (!ret) { return false; }
+
+		var is_key = Module.ccall('is_keyframe', 'boolean', ['number'], [frame_idx]);
+
+		// Positions - fetch and upload.
+		var vp_copied = Module.ccall('frame_vp_copied', 'number');
+		var vp_sz = Module.ccall('frame_vertices_sz', 'number');
+		var vp_f32 = new Float32Array(Module.HEAP8.buffer, vp_copied, vp_sz / 4);
+		vologram.geometry.setAttribute( 'position', new THREE.Float32BufferAttribute(vp_f32, 3 ));
+
+		if (vologram.has_normals) { // Not all volograms include normals.
+			// Normals - fetch and upload.
+			var normals_copied = Module.ccall('frame_normals_copied', 'number');
+			var normals_sz = Module.ccall('frame_normals_sz', 'number');
+			var vn_f32 = new Float32Array(Module.HEAP8.buffer, normals_copied, normals_sz / 4);
+			vologram.geometry.setAttribute( 'normal', new THREE.Float32BufferAttribute(vn_f32, 3 ));
+		}
+
+		// Key-Frames also contain texture coordinate and index data.
+		if (is_key) {
+			vologram.last_keyframe_loaded = frame_idx;
+
+			// Texture Coordinates - fetch and upload.
+			var uvs_copied = Module.ccall('frame_uvs_copied', 'number');
+			var uvs_sz = Module.ccall('frame_uvs_sz', 'number');
+			var uvs_f32 = new Float32Array(Module.HEAP8.buffer, uvs_copied, uvs_sz / 4);
+			vologram.geometry.setAttribute( 'uv', new THREE.Float32BufferAttribute( uvs_f32, 2 ));
+
+			// Indices - fetch and upload.
+			var indices_copied = Module.ccall('frame_indices_copied', 'number');
+			var indices_sz = Module.ccall('frame_i_sz', 'number');
+			vologram.n_indices = indices_sz / 2; // ushort
+			var indices_u16 = new Uint16Array(Module.HEAP8.buffer, indices_copied, vologram.n_indices);
+			vologram.geometry.setIndex( new THREE.Uint16BufferAttribute( indices_u16, 1 ))
+		}
+
+		// It seems that calculating bounding sphere does not work and always returns NaN value for the radius
```

Comment:
i bet the NaN is when there is nobody in the scene. could probably check the vertex count > 0 before doing the bounding sphere update.

Reply:
For this example, we don't need the bounding sphere, but I don't know if its NaN because of a vologram issue, or a threejs issue